packages/odyssey-react-mui/src/theme/useContrastMode.test.tsx

```diff
@@ -0,0 +1,254 @@
+
+// Common wrapper component for tests
+const createWrapper = (contextValue: ContrastModeContextType) => {
+  return ({ children }: { children: React.ReactNode }) => (
+    <ContrastModeContext.Provider value={contextValue}>
+      {children}
+    </ContrastModeContext.Provider>
+  );
+};
+
+describe("useContrastMode and related functions", () => {
+  const originalGetComputedStyle = window.getComputedStyle;
+
+  beforeEach(() => {
+    window.getComputedStyle = jest.fn().mockImplementation(() => ({
+      backgroundColor: "rgba(0, 0, 0, 0)",
+    }));
+  });
+
+  afterEach(() => {
+    window.getComputedStyle = originalGetComputedStyle;
+    jest.clearAllMocks();
+    document.documentElement.style.backgroundColor = "";
+  });
+
+  describe("useContrastMode hook", () => {
+    it("should return lowContrast mode by default", () => {
+      const { result } = renderHook(() => useContrastMode({}), {
+        wrapper: createWrapper({ contrastMode: "lowContrast" }),
+      });
+
+      expect(result.current.contrastMode).toBe("lowContrast");
+      expect(result.current.contrastContainerRef.current).toBe(null);
+      expect(result.current.parentBackgroundColor).toBe("#ffffff");
+    });
+
+    it("should respect explicitly set contrast mode", () => {
+      const { result } = renderHook(
+        () => useContrastMode({ contrastMode: "highContrast" }),
+        {
+          wrapper: createWrapper({ contrastMode: "lowContrast" }),
+        },
+      );
+
+      expect(result.current.contrastMode).toBe("highContrast");
+    });
+
+    it("should update contrast mode based on background color changes", async () => {
+      const TestComponent = () => {
+        const { contrastContainerRef, contrastMode } = useContrastMode({});
+        return (
+          <div ref={contrastContainerRef} data-testid="container">
+            {contrastMode}
+          </div>
+        );
+      };
+
+      // Mock getComputedStyle before rendering
+      window.getComputedStyle = jest.fn().mockImplementation(() => ({
+        backgroundColor: Tokens.HueNeutral50,
+      }));
+
+      const { getByTestId } = render(
+        <ContrastModeContext.Provider value={{ contrastMode: "lowContrast" }}>
+          <TestComponent />
+        </ContrastModeContext.Provider>,
+      );
+
+      const testContainer = getByTestId("container");
+
+      act(() => {
+        testContainer.style.backgroundColor = Tokens.HueNeutral50;
+        const event = new Event("transitionend");
+        Object.defineProperty(event, "propertyName", {
+          value: "background-color",
+        });
+        testContainer.dispatchEvent(event);
+      });
+
+      // Add a small delay to allow for state updates
+      await act(async () => {
+        await new Promise((resolve) => setTimeout(resolve, 0));
+      });
+
+      await waitFor(
+        () => {
+          expect(getByTestId("container").textContent).toBe("highContrast");
+        },
+        { timeout: 1000 },
+      );
```

Comment:
I wonder if we still need this explicit timeout. No issue if we have it, just wanna make the test simpler.

Reply:
good call, I tested without it and it isn't needed 🎉